src/nl/hannahsten/texifyidea/settings/conventions/TexifyConventionsConfigurable.kt

```diff
@@ -0,0 +1,233 @@
+                            value: Any?,
+                            selected: Boolean,
+                            focus: Boolean,
+                            row: Int,
+                            column: Int
+                        ): Component {
+                            val convention = value as LabelConvention?
+                            super.getTableCellRendererComponent(table, convention?.name, selected, focus, row, column)
+                            //noinspection unchecked
+                            icon = if (value != null) {
+                                when (value.type!!) {
+                                    LabelConventionType.ENVIRONMENT -> TexifyIcons.DOT_ENVIRONMENT
+                                    LabelConventionType.COMMAND -> TexifyIcons.DOT_COMMAND
+                                }
+                            }
+                            else {
+                                null
+                            }
+
+                            return this
+                        }
+                    }
+                }
+            }
+
+        val enabledColumnInfo = object : TableModelEditor.EditableColumnInfo<LabelConvention, Boolean>("") {
+            override fun getColumnClass(): Class<*> = Boolean::class.java
+
+            override fun valueOf(item: LabelConvention): Boolean = item.enabled
+            override fun setValue(item: LabelConvention, value: Boolean) {
+                item.enabled = value
+            }
+        }
+
+        val itemEditor = object : CollectionItemEditor<LabelConvention> {
+            override fun getItemClass(): Class<out LabelConvention> = LabelConvention::class.java
+
+            override fun clone(item: LabelConvention, forInPlaceEditing: Boolean): LabelConvention {
+                TODO("Not yet implemented")
+            }
+
+            override fun isRemovable(item: LabelConvention): Boolean = false
+        }
+
+        labelConventionsTable = object : TableModelEditor<LabelConvention>(
+            listOf(enabledColumnInfo, nameColumnInfo, prefixColumnInfo).toTypedArray(),
+            itemEditor,
+            "Label Conventions"
+        ) {
+            override fun canCreateElement(): Boolean = false
+        }
+        labelConventionsTable.disableUpDownActions()
+
+        maxSectionSize = JBIntSpinner(4000, 100, Integer.MAX_VALUE)
+        val centerPanel = panel {
+            row {
+                label("Maximum section size (characters)")
+                maxSectionSize(grow)
+            }
+
+            titledRow("Labels") {
+                row {
+                    labelConventionsTable.createComponent()(grow)
+                }
+            }
+        }
+
+        mainPanel = JPanel(BorderLayout())
+        mainPanel.add(schemesPanel, BorderLayout.NORTH)
+        mainPanel.add(centerPanel, BorderLayout.CENTER)
+        loadSettings(settingsManager.getSettings())
+        return mainPanel
```

Comment:
Now, we should carefully consider which settings should be on this page. I think the "minimum section level which triggers the missing label inspection" should be here, because it recommends certain LaTeX conventions, and other settings can remain where they are.

Reply:
The reason I didn't include the "minimum section level which triggers the missing label inspection" setting was because of my intended meaning of the checkbox (see my explanation above).

The `sectionSeparatorColors` would be an possible candidate for the conventions though. Alternatively we could inlcude them in the "Editor -> Color Scheme" like other languages do.

Reply:
Yes, that makes my comment redundant (except that we shouldn't forget to remove that setting there). 
I would think colors should be in the color scheme, yes.